### ISSUE TITLE
Train and Validation predictions vs Testing Predictions Discrepancy

### ISSUE THREAD

**Issue by selkind** [selkind/LaSeRo#9]
The models trained today
 - transfer_burjobands_lr_0.001_val_loss
 - transfer_burjobands_lr_0.001_val_accuracy
- one_percent_burjobands_lr_0.001_*
all achieved validation and training accuracy in the mid-high 90% range.
Test-set predictions are terrible.
Most of the test chunk predictions are almost entirely labeled "rock" with some not-rock random noise.
If models were following this pattern in the training/validation set, accuracy would be close to zero due to the scarcity of rocks in the images.

What is the cause of this contradiction?

**Comment by selkind** [comment]
I have two Hypotheses:
First: The two things that changed in the model today were 
- the to_categorical() got a num_classes argument.
- The visualization functions got refactored into a utility module.

Either or both of these changes could have led to unintended changes in the labels created by the model predictions or loaded into the training/validation process.

**Comment by selkind** [comment]
Second:
The testing images are sourced from the qc directory shared by ABJ. These are corrected in such a way that the values are sufficiently different to be incomparable with the training data and therefore incompatible with the model.

**Comment by selkind** [comment]
>
> First: The two things that changed in the model today were
> 
>     * the to_categorical() got a num_classes argument.
> 
>     * The visualization functions got refactored into a utility module.
To test this I will:

- [ ] Manually verify the output of the to_categorical() function with and without the num_classes arg
- [ ] Step through the diff and visualization code to ensure they're consistent with the previous implementation

**Comment by selkind** [comment]
> 
> 
> Second:
> The testing images are sourced from the qc directory shared by ABJ. These are corrected in such a way that the values are sufficiently different to be incomparable with the training data and therefore incompatible with the model.

To test this I will:
- [ ] compare histograms of the test_chunks arrays with the chunks in the qc directory. They should be different. If they are not, we need to re-make the test_chunks directory, sourcing from the less corrected imagery accessible by Amazon S3 buckets

**Comment by selkind** [comment]
This issue was occurring because the predictor was generating results on bands that were different from the bands that the model was trained on. This has been fixed as of commit 5e51490.

Band information is now included in the naming schema of the model directories and can be parsed to be used as a variable for the predictor notebook
